sky/authentication.py

```diff
@@ -267,10 +268,31 @@ def setup_gcp_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
     return _replace_ssh_info_in_config(config, public_key)
 
 
+# Cloud init script is encoded in base64 in azure, so we need to decode it
+# and replace the ssh user and public key content, then encode it back.
```

Comment:
Do we really need to encode the script with base64?

Reply:
Yes... Azure only accepts base64 encoded in the `customData` field. See https://github.com/skypilot-org/skypilot/pull/1910#discussion_r1262930925 for more information

Reply:
Let me add a comment for this